lgc/patch/LowerDebugPrintf.cpp

```diff
@@ -90,10 +90,9 @@ void LowerDebugPrintf::visitDebugPrintf(DebugPrintfOp &op) {
   SmallVector<Value *> printArgs;
   // Records printf output variables are 64bit or not
   SmallBitVector bit64Vector;
-  for (Value *var : op.getArgs()) {
```

Comment:
The windows compilation error is probably because the iterator type on Linux is a pointer, but on Windows it’s a different type.
Can you use `auto var : op.getArgs()`?

Reply:
auto is Value& on windows. Value* on linux

we still need to get the different var to the 
getDwordValues

Reply:
If it’s really such different types (which makes `getArgs` unusable), this needs to be fixed in llvm-dialects instead of working around it here.

Reply:
> If it’s really such different types (which makes `getArgs` unusable), this needs to be fixed in llvm-dialects instead of working around it here.

right. i am not familiar with llvm-dialects.   from LgcDialect.cpp.inc
i think 
llvm::iterator_range<llvm::Value*> should be llvm::iterator_range<User::op_iterator>

but i don't know how  to correct this in llvm-dialect.

so i figure having a fix is better than no fix.

Reply:
llvm::iterator_range is very simple implementation with begin and end().  so it must use with op_iterator.
So even the linux can compile and run. i guess something could be wrong with many var aguments.

Reply:
Has the local checkout of the llvm-dialects submodule been updated? Maybe it's still using an old version, as this problem was fixed recently...

If the submodule *has* been updated, maybe there's a missing dependency in the cmake?

Reply:
Presumably the change went in to the GPUOpen llvm-dialects (since that's the definitive version)?